poetry/console/commands/env/remove.py

```diff
@@ -1,21 +1,44 @@
 from cleo.helpers import argument
+from cleo.helpers import option
 
 from ..command import Command
 
 
 class EnvRemoveCommand(Command):
 
     name = "env remove"
-    description = "Removes a specific virtualenv associated with the project."
+    description = "Remove specific virtual environments associated with the project."
 
     arguments = [
-        argument("python", "The python executable to remove the virtualenv for.")
+        argument(
+            "python",
+            "The python executables or names of the virtual environments which are to "
+            "be removed.",
+            optional=True,
+            multiple=True,
+        )
+    ]
+    options = [
+        option(
+            "all",
+            description="Remove all managed virtual environments associated with the "
+            "project.",
+        ),
     ]
 
     def handle(self) -> None:
         from poetry.utils.env import EnvManager
 
-        manager = EnvManager(self.poetry)
-        venv = manager.remove(self.argument("python"))
+        pythons = self.argument("python")
+        all = self.option("all")
+        if not (pythons or all):
+            self.line("No virtualenv provided.")
 
-        self.line("Deleted virtualenv: <comment>{}</comment>".format(venv.path))
+        manager = EnvManager(self.poetry)
+        for python in pythons:
```

Comment:
```suggestion
        # TODO: refactor env.py to allow removal with one loop
        for python in pythons:
```

Let's drop a `TODO` here to clean this up a little better.